src/transformers/models/rt_detr/modeling_rt_detr.py

```diff
@@ -1310,13 +1309,13 @@ def __init__(self, config: RTDetrConfig):
     def build_2d_sincos_position_embedding(
         width, height, embed_dim=256, temperature=10000.0, device="cpu", dtype=torch.float32
     ):
-        grid_w = torch.arange(int(width), dtype=dtype, device=device)
-        grid_h = torch.arange(int(height), dtype=dtype, device=device)
+        grid_w = torch.arange(int(width), device=device).to(dtype)
+        grid_h = torch.arange(int(height), device=device).to(dtype)
```

Comment:
The `Range` op in fp16 is not supported by ONNX, so we use it with the default type and then cast it to the desired data type.